modules/images/dominant-color/load.php

```diff
@@ -417,3 +417,1432 @@ function dominant_color_rgb_to_hex( $red, $green, $blue ) {
 
 	return sprintf( '%02x%02x%02x', $red, $green, $blue );
 }
+
+/**
+ * Integrates with the media library.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_add_media_library_integration() {
+	add_action( 'admin_footer', 'dominant_color_print_media_templates' );
+	add_filter( 'wp_prepare_attachment_for_js', 'dominant_color_prepare_attachment_for_js', 10, 3 );
+}
+add_filter( 'plugins_loaded', 'dominant_color_add_media_library_integration' );
+remove_action( 'admin_footer', 'wp_print_media_templates' );
+
+/**
+ * Add the dominant color when preparing attachment data.
+ *
+ * @since n.e.x.t
+ *
+ * @param array       $response   Array of prepared attachment data. @see wp_prepare_attachment_for_js().
+ * @param WP_Post     $attachment Attachment object.
+ * @param array|false $meta       Array of attachment meta data, or false if there is none.
+ * @return array Array of prepared attachment data.
+ */
+function dominant_color_prepare_attachment_for_js( $response, $attachment, $meta ) {
+	if ( ! wp_attachment_is_image( $attachment->ID ) ) {
+		return $response;
+	}
+
+	$dominant_color = dominant_color_get_dominant_color( $attachment->ID );
+	if ( ! $dominant_color ) {
+		return $response;
+	}
+
+	$response['dominantColor'] = $dominant_color;
+
+	// Set hasTransparency.
+	if ( dominant_color_has_transparency( $attachment->ID ) ) {
+		$response['hasTransparency']      = 'true';
+		$response['hasTransparencyClass'] = 'has-transparency';
+	} else {
+		$response['hasTransparency']      = 'false';
+		$response['hasTransparencyClass'] = 'no-transparency';
+	}
+
+	return $response;
+}
+
+/**
+ * Prints the templates used in the media manager.
+ *
+ * Copied from core's wp_print_media_templates() and modified to add the dominant color.
+ *
+ * Eslint is disabled because this block of code is copied from core and only included here to enable modification.
+ *
+ * @since n.e.x.t
+ */
+// phpcs:disable
+ function dominant_color_print_media_templates() {
+	$class = 'media-modal wp-core-ui';
+
+	$alt_text_description = sprintf(
+		/* translators: 1: Link to tutorial, 2: Additional link attributes, 3: Accessibility text. */
+		__( '<a href="%1$s" %2$s>Learn how to describe the purpose of the image%3$s</a>. Leave empty if the image is purely decorative.' ),
+		esc_url( 'https://www.w3.org/WAI/tutorials/images/decision-tree' ),
+		'target="_blank" rel="noopener"',
+		sprintf(
+			'<span class="screen-reader-text"> %s</span>',
+			/* translators: Accessibility text. */
+			__( '(opens in a new tab)' )
+		)
+	);
+	?>
+
+	<?php
+	// Template for the media frame: used both in the media grid and in the media modal.
+	// Include the inline styles for the dominant color feature.
+	?>
+	<style>
+		.wp-core-ui .attachment-preview[data-dominant-color]:not(.has-transparency) {
+			background: var(--dominant-color);
```

Comment:
Instead of `background` use `background-color` so it will not override other background properties. For the frontend request, we used `background-color` https://github.com/WordPress/performance/blob/trunk/modules/images/dominant-color/load.php#L243.

```suggestion
			background-color: var(--dominant-color);
```